core/arch/riscv/include/kernel/thread_private_arch.h

```diff
@@ -0,0 +1,71 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2022 NXP
+ */
+
+#ifndef __KERNEL_THREAD_PRIVATE_ARCH_H
+#define __KERNEL_THREAD_PRIVATE_ARCH_H
+
+#ifndef __ASSEMBLER__
+
+#include <kernel/thread.h>
+
+#define STACK_TMP_OFFS		0
+
+#if defined(__clang__) && !defined(__OPTIMIZE_SIZE__)
+#define STACK_TMP_SIZE		(4096 + STACK_TMP_OFFS + CFG_STACK_TMP_EXTRA)
+#else
+#define STACK_TMP_SIZE		(2048 + STACK_TMP_OFFS + CFG_STACK_TMP_EXTRA)
+#endif
+
+#define STACK_THREAD_SIZE	(8192 + CFG_STACK_THREAD_EXTRA)
+
+#if TRACE_LEVEL > 0
+#define STACK_ABT_SIZE		3072
+#else
+#define STACK_ABT_SIZE		1024
+#endif
+
+#ifdef CFG_CORE_DEBUG_CHECK_STACKS
+#define STACK_CHECK_EXTRA	1536
```

Comment:
It seems that one needs to read the Arm code in order to understand the purpose of this. If you don't use this keep it as `0` for simplicity.